bin/etest

```diff
@@ -230,77 +230,59 @@ die_handler()
     exit ${rc}
 }
 
-# Returns success if there are no stale processes remaining in the cgroup for this test and failure if there are any.
-no_process_leaks_remain()
+assert_no_process_leaks()
 {
     if ! cgroup_supported; then
         return 0
     fi
 
-    # If the cgroup no longer exists, we're in good shape because you can't destroy a cgroup until all its processes are
-    # dead.
-    $(tryrc -r=exists_rc cgroup_exists ${ETEST_CGROUP})
-    if [[ ${exists_rc} -ne 0 ]] ; then
+    edebug "Checking for process leaks in ${ETEST_CGROUP}"
+
+    if ! cgroup_exists "${ETEST_CGROUP}"; then
+        edebug "$(lval ETEST_CGROUP) does not exist -- returning"
         return 0
     fi
 
-    # Get a list of pids in our CGROUP, excluding the elogfile pids as we know they are always going to be running at
-    # this point.
-    #
-    # NOTE: Since we just checked that the cgroup exists, we also know the cgroup_pids will exist, because nothing else
-    #       will destroy the cgroup except for us.
-    local remaining_pids=""
-    remaining_pids=$(cgroup_pids --exclude "$(elogfile_pids)" ${ETEST_CGROUP})
-    etestmsg "$(lval remaining_pids exists_rc ETEST_CGROUP)"
-    [[ -z ${remaining_pids} ]]
-}
-
-assert_no_process_leaks()
-{
-    # Error stacks generated here should produce output, even though etest has them wrapped in a try.
-    __EBASH_INSIDE_TRY=0
-
-    edebug "Waiting..."
+    __check_process_leaks()
+    {
+        [[ -z "$(cgroup_pids "${ETEST_CGROUP}")" ]]
+    }
```

Comment:
This private function local to this function is what replaces `no_process_leaks_remain`